### PR TITLE
changing templates for coherence

### PR DESCRIPTION
**Pull request by orklah**
This PR propose changing some templates annotations for psalm.

Most of the time, it shouldn't be needed to add local @var with templates, it means there is something weird going on.

In the case of ProjectDescriptorBuilder.php, it was just an omission in getAssembler documentation.

The AssemblerFactory.php is more complex though. If I understand it fully, The class is a collection of AssemblerMatcher built in advance. When needed, we send a Descriptor, and it returns a matching AssemblerInterface.

That means the Descriptor given and the one retrieved is not actually the same and Psalm detects it. We know it's basically the same because it matched, so I had to cheat a little with an annotation saying that the returned element is the one given in parameter.

### CODE REVIEW

**Comment by orklah** [comment]
I only checked with psalm. Phpstan is failing, I'll look into it

**Comment by orklah** [comment]
The remaining error is a bug in phpstan: https://github.com/phpstan/phpstan/issues/3951

I'll try a workaround if this is not fixed in a few days

File: phpstan.neon [hunk 1]
```diff
@@ -1,5 +1,4 @@
 includes:
-    - /composer/vendor/phpstan/phpstan-webmozart-assert/extension.neon
```
Comment (mvriel) [inline phpstan.neon:2]: Why did you remove this extension? Is it not needed for the assertion support?

Comment (orklah) [reply phpstan.neon:2]: arg, shouldn't have included that in the commit

Comment (orklah) [reply phpstan.neon]: I remove it locally because the path is not valid, but I revert that before committing usually.

I'm a little surprised it didn't seem to cause issue with the build though...

I reverted it

**Comment by orklah** [comment]
I ignored the issue for now, this should be ready

**Comment by jaapio** [comment]
Thanks for cleaning up this part
